src/synology_dsm/api/photos/__init__.py

```diff
@@ -70,17 +78,22 @@ async def get_items_from_album(
         self, album: SynoPhotosAlbum, offset: int = 0, limit: int = 100
     ) -> list[SynoPhotosItem] | None:
         """Get a list of all items from given album."""
+        params = {
+            "offset": offset,
+            "limit": limit,
+            "additional": '["thumbnail"]',
+        }
+        if album.passphrase is not None:
+            params["passphrase"] = album.passphrase
+        else:
+            params["album_id"] = album.album_id
```

Comment:
is the album ID really not needed when we have a passphrase?

Reply:
Yes it is, requesting a shared album with both keys gives a "120 Invalid parameter" response.
Also, that's what the web client does.